### PR TITLE
feat: Support python 3.10 runtime in apps

### PR DESCRIPTION
**Pull request by VijithaEkanayake**
Fixes: https://github.com/h2oai/wave-apps/issues/125

### CODE REVIEW

File: churn-risk/requirements.txt [hunk 1]
```diff
@@ -1,8 +1,9 @@
-pandas==1.1.0
-pygments==2.7.1
+pandas==1.4.0
+Pygments==2.15.0
```
Comment (mnezh) [inline churn-risk/requirements.txt:2]: newer pygments? https://pypi.org/project/Pygments/2.17.2/

File: credit-risk/requirements.txt [hunk 2]
```diff
@@ -1,11 +1,12 @@
-numpy==1.19.1
-pandas==1.1.0
+numpy==1.22.2
```
Comment (mnezh) [inline credit-risk/requirements.txt:1]: https://pypi.org/project/numpy/1.26.4/

File: credit-risk/requirements.txt [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
+pandas==1.4.0
```
Comment (mnezh) [inline credit-risk/requirements.txt:2]: https://pypi.org/project/pandas/1.5.3/

File: credit-risk/requirements.txt [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
 plotly==4.10.0
-pygments==2.7.1
+pygments==2.15.0
```
Comment (mnezh) [inline credit-risk/requirements.txt:4]: https://pypi.org/project/Pygments/2.17.2/

File: explaining-ratings/requirements.txt [hunk 5]
```diff
@@ -1,4 +1,6 @@
-pandas==1.1.0
+pandas==1.4.0
```
Comment (mnezh) [inline explaining-ratings/requirements.txt:1]: https://pypi.org/project/pandas/1.5.3/

File: explaining-ratings/requirements.txt [hunk 6]
```diff
@@ -1,4 +1,6 @@
...
 h2o-wave<1.0
 matplotlib
-wordcloud
\ No newline at end of file
+wordcloud
```
Comment (mnezh) [inline explaining-ratings/requirements.txt:4]: let's freeze the version https://pypi.org/project/wordcloud/1.9.3/

File: explaining-ratings/requirements.txt [hunk 7]
```diff
@@ -1,4 +1,6 @@
-pandas==1.1.0
+pandas==1.4.0
 h2o-wave<1.0
 matplotlib
```
Comment (mnezh) [inline explaining-ratings/requirements.txt:3]: let's freeze the version https://pypi.org/project/matplotlib/3.8.3/

File: credit-risk/requirements.txt [hunk 8]
```diff
@@ -1,11 +1,12 @@
...
 h2o-wave<1.0
 h2o-wave-ml>=0.7.0
 requests
```
Comment (mnezh) [inline credit-risk/requirements.txt:7]: let's freeze the version for predictable deployments

File: credit-risk/requirements.txt [hunk 9]
```diff
@@ -1,11 +1,12 @@
...
 tabulate
```
Comment (mnezh) [inline credit-risk/requirements.txt:8]: let's freeze the version for predictable deployments

File: credit-risk/requirements.txt [hunk 10]
```diff
@@ -1,11 +1,12 @@
...
 colorama>=0.3.8
 future
```
Comment (mnezh) [inline credit-risk/requirements.txt:10]: let's freeze the version for predictable deployments

File: explaining-ratings/requirements.txt [hunk 11]
```diff
@@ -1,4 +1,6 @@
...
-wordcloud
\ No newline at end of file
+wordcloud
+numpy==1.22.2
```
Comment (mnezh) [inline explaining-ratings/requirements.txt:5]: https://pypi.org/project/numpy/1.26.4/

File: insurance-churn-risk/requirements.txt [hunk 12]
```diff
@@ -1,7 +1,10 @@
-pandas==1.1.0
-pygments==2.7.1
+certifi==2024.2.2
+pandas==1.4.0
+pygments==2.15.0
 h2o-wave<1.0
 h2o-wave-ml>=0.6.0
+numpy==1.22.2
```
Comment (mnezh) [inline insurance-churn-risk/requirements.txt:6]: https://pypi.org/project/numpy/1.26.4/

File: insurance-churn-risk/requirements.txt [hunk 13]
```diff
@@ -1,7 +1,10 @@
-pandas==1.1.0
-pygments==2.7.1
+certifi==2024.2.2
+pandas==1.4.0
```
Comment (mnezh) [inline insurance-churn-risk/requirements.txt:2]: let's use newer

File: credit-risk/requirements.txt [hunk 14]
```diff
@@ -1,11 +1,12 @@
-numpy==1.19.1
-pandas==1.1.0
+numpy==1.22.2
+pandas==1.4.0
 plotly==4.10.0
-pygments==2.7.1
+pygments==2.15.0
 h2o-wave<1.0
```
Comment (mnezh) [inline credit-risk/requirements.txt:5]: https://pypi.org/project/h2o-wave/1.0.2/

**Review by mnezh — CHANGES_REQUESTED** [review]

File: template-explore-binary-classification/requirements.txt [hunk 15]
```diff
@@ -1,5 +1,11 @@
-h2o_wave==0.20.0
+certifi==2024.2.2
+h2o_wave==0.26.3
```
[the remark this answers is not in the document]
Comment (VijithaEkanayake) [reply template-explore-binary-classification/requirements.txt:2]: Wave 1.0 introduces a breaking change for handle_on functionality. So Wave update needs to be handled separately.

File: insurance-churn-risk/requirements.txt [hunk 16]
```diff
@@ -1,7 +1,10 @@
...
+pygments==2.15.0
 h2o-wave<1.0
```
[the remark this answers is not in the document]
Comment (VijithaEkanayake) [reply insurance-churn-risk/requirements.txt:4]: Wave 1.0 introduces a breaking change for handle_on functionality. So Wave update needs to be handled separately.

File: guess-the-number/requirements.txt [hunk 17]
```diff
@@ -1,12 +1,12 @@
-certifi==2020.12.5; python_version >= "3.6" and python_full_version >= "3.6.1"
-click==7.1.2; python_full_version >= "3.6.1"
-h11==0.11.0; python_version >= "3.6" and python_full_version >= "3.6.1"
+certifi==2024.2.2
+click==7.1.2
+h11==0.11.0
 h2o-wave<1.0
```
[the remark this answers is not in the document]
Comment (VijithaEkanayake) [reply guess-the-number/requirements.txt:4]: Wave 1.0 introduces a breaking change for handle_on functionality. So Wave update needs to be handled separately.

File: churn-risk/requirements.txt [hunk 18]
```diff
@@ -1,8 +1,9 @@
...
 h2o-wave<1.0
```
[the remark this answers is not in the document]
Comment (VijithaEkanayake) [reply churn-risk/requirements.txt:3]: Wave 1.0 introduces a breaking change for handle_on functionality. So Wave update needs to be handled separately.

File: credit-risk/requirements.txt (hunk 14 above)
Comment (VijithaEkanayake) [reply credit-risk/requirements.txt:5]: Wave 1.0 introduces a breaking change for handle_on functionality. So Wave update needs to be handled separately.

File: sales-forecasting/requirements.txt [hunk 19]
```diff
@@ -1,22 +1,22 @@
 boto3==1.16.23
 botocore==1.19.23
-certifi==2020.11.8; python_version >= "3.6" and python_full_version >= "3.6.1"
+certifi==2024.2.2
 click==7.1.2; python_full_version >= "3.6.1"
 h11==0.11.0; python_version >= "3.6" and python_full_version >= "3.6.1"
 h2o-wave<1.0; python_full_version >= "3.6.1"
```
[the remark this answers is not in the document]
Comment (VijithaEkanayake) [reply sales-forecasting/requirements.txt:6]: Wave 1.0 introduces a breaking change for handle_on functionality. So Wave update needs to be handled separately.

**Review by mnezh — APPROVED** [review]
agree on wave 1.0+
lgtm

**Review by mnezh — APPROVED** [review]

**Review by mturoci — APPROVED** [review]

**Pull request merged**